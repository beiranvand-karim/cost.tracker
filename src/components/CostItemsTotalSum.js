import React from "react";

function CostItemsTotalSum({ costItems }) {
  const sum = costItems
    .map(costItem => costItem.cost)
    .reduce((total, cost) => total + Number(cost), 0);

  return <div>total sum of transactions: {sum}</div>;
}

export default CostItemsTotalSum;
