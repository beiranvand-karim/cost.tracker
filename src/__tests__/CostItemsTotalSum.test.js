/* global expect, describe, it, beforeEach */
import CostItemsTotalSum from "../components/CostItemsTotalSum";
import costItems from "../__helpers__/costItems";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import React from "react";

describe("<CostItemsTotalSum/>", () => {
  it("should render without crashing", () => {
    shallow(<CostItemsTotalSum costItems={costItems} />);
  });

  it("should matches the snapshot", () => {
    const tree = renderer
      .create(<CostItemsTotalSum costItems={costItems} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
