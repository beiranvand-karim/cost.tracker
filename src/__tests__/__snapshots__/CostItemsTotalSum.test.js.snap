// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CostItemsTotalSum/> should matches the snapshot 1`] = `
<div>
  total sum of transactions: 
  4500
</div>
`;
